numba/cpython/listobj.py

```diff
@@ -1119,3 +1120,85 @@ def list_to_list(context, builder, fromty, toty, val):
     # Casting from non-reflected to reflected
     assert fromty.dtype == toty.dtype
     return val
+
+# -----------------------------------------------------------------------------
+# Implementations for types.LiteralList
+# -----------------------------------------------------------------------------
+
+# Things that mutate literal lists are banned
+@overload_method(types.LiteralList, 'append')
+@overload_method(types.LiteralList, 'extend')
+@overload_method(types.LiteralList, 'insert')
+@overload_method(types.LiteralList, 'remove')
+@overload_method(types.LiteralList, 'pop')
+@overload_method(types.LiteralList, 'clear')
+@overload_method(types.LiteralList, 'sort')
+@overload_method(types.LiteralList, 'reverse')
+def banned_mutators(*args):
+    # heterogeneous lists are a trick, they are really tuples
+    raise errors.TypingError("Cannot mutate a literal list")
+
+_index_end = types.intp.maxval
+@overload_method(types.LiteralList, 'index')
+def literal_list_index(lst, x, start=0, end=_index_end):
+    # TODO: To make this work, need consts as slice for start/end so as to
+    # be able to statically analyse the bounds, then its a just loop body
+    # versioning based iteration along with enumerate to find the item
+    if isinstance(lst, types.LiteralList):
+        msg = "list.index is unsupported for literal lists"
+        raise errors.TypingError(msg)
+
+@overload_method(types.LiteralList, 'count')
+def literal_list_count(lst, x):
+    if isinstance(lst, types.LiteralList):
+        def impl(lst, x):
+            count = 0
+            for val in literal_unroll(lst):
+                if val == x:
+                    count += 1
+            return count
+        return impl
+
+@overload_method(types.LiteralList, 'copy')
+def literal_list_count(lst):
+    if isinstance(lst, types.LiteralList):
+        def impl(lst):
+            return lst # tuples are immutable, as is this, so just return it
+        return impl
+
+@overload(operator.delitem)
+def literal_list_delitem(lst, index):
+    if isinstance(lst, types.LiteralList):
+        raise errors.TypingError("Cannot mutate a literal list")
+
+@overload(operator.setitem)
+def literal_list_setitem(lst, index, value):
+    if isinstance(lst, types.LiteralList):
+        raise errors.TypingError("Cannot mutate a literal list")
+
+@overload(operator.getitem)
+def literal_list_getitem(lst, *args):
+    if not isinstance(lst, types.LiteralList):
+        return
+    msg = ("Cannot __getitem__ on a literal list, return type cannot be "
+           "statically determined.")
+    raise errors.TypingError(msg)
+
+@overload(len)
+def literal_list_len(lst):
+    if not isinstance(lst, types.LiteralList):
+        return
+    llen = len(lst.types)
```

Comment:
never used. because [constsized_len](https://github.com/numba/numba/blob/e043ee705d65abd3a066500be4d62b54bdf3586c/numba/cpython/builtins.py#L367) takes precedence.